### PR TITLE
Bug/ux 540 create form styling extra fields

### PR DESCRIPTION
**Pull request by AllenBW**

### CODE REVIEW

File: app/assets/javascripts/admin/products/product_form_controller.js [hunk 1]
```diff
@@ -1,32 +1,75 @@
...
   var self = this;
 
   this.product = null;
   this.formSubmitted = false;
+  $scope.prodsByName = ProductTypesResource.query();
+
+
+  $scope.$watchCollection('productForm.product.product_type',
+    function(productTypeSelected) {
+    angular.forEach($scope.prodsByName, function(item){
```
Comment (stackus) [inline app/assets/javascripts/admin/products/product_form_controller.js]: There is a chance that the promise has not returned a value yet and this value will not have an enumerable list.

File: app/assets/javascripts/admin/products/product_form_controller.js [hunk 2]
```diff
@@ -1,32 +1,75 @@
...
+                               FlashesService, ProductTypesResource) {
+  'use strict';
   var self = this;
 
   this.product = null;
   this.formSubmitted = false;
+  $scope.prodsByName = ProductTypesResource.query();
+
+
+  $scope.$watchCollection('productForm.product.product_type',
```
Comment (stackus) [inline app/assets/javascripts/admin/products/product_form_controller.js]: This looks like something ngChange on the select element should be able to do.

File: app/assets/javascripts/admin/products/product_form_controller.js [hunk 3]
```diff
@@ -1,32 +1,75 @@
...
+        disableSuccessState: true}]
+      });
+    });
+    this.schemaForm = self.formItems;
 
-  this.priceRegex = "\\d{1,6}(\\.\\d{1,4})?";
+    return false;
+  };
+
+  this.priceRlmegex = '\\d{1,6}(\\.\\d{1,4})?';
```
Comment (stackus) [inline app/assets/javascripts/admin/products/product_form_controller.js]: 'Rlm'?

File: app/assets/javascripts/admin/products/product_form_controller.js [hunk 4]
```diff
@@ -1,53 +1,96 @@
...
+        disableSuccessState: true}]
+      });
+    });
+    this.schemaForm = self.formItems;
 
-  this.priceRegex = "\\d{1,6}(\\.\\d{1,4})?";
+    return false;
+  };
+
+  this.priceRlmegex = "\\d{1,6}(\\.\\d{1,4})?";
```
Comment (AllenBW) [inline app/assets/javascripts/admin/products/product_form_controller.js]: yep shouldn't be there, good catch.

File: app/assets/javascripts/admin/products/product_form_controller.js [hunk 5]
```diff
@@ -5,49 +5,91 @@
...
+  $scope.prodsByName = ProductTypesResource.query();
+
+
+  $scope.$watchCollection('productForm.product.product_type', function(productTypeSelected) {
+    angular.forEach($scope.prodsByName, function(item){
+      if(item.title == productTypeSelected){
+        self.formItems = [];
+        angular.forEach(item.properties, function (item, key) {
+          self.formItems.push({
+            type: "section",
```
Comment (AllenBW) [inline app/assets/javascripts/admin/products/product_form_controller.js]: perhaps, though my initial implementation which used an ng-click  threw capybara into a digest tizzy (ie wouldn't pass the build) so i moved all the work into the controller.

**Pull request merged**